packages/app/features/board/index.js

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import "./style.css"
+
+function Square({ value, onSquareClick }) {
+    return (
+        <button
+            className="square"
+            onClick={onSquareClick}
+        >
+            {value}
+        </button>
+    );
+}
+
+export default function Board() {
+
+    let[xTurn, setXTurn] = useState(true);
+    
+    let [sqrValues, setSqrValues] = useState(Array(9).fill(null));
+
+    function handleClick(i) {
+        if(sqrValues[i] || theWinnerIs(sqrValues)) { return; }
+
+        let nextValues = sqrValues.slice();
+        
+        if (xTurn) {
+            nextValues[i] = "X";
+        } else {
+            nextValues[i] = "O";
+        }
+
+        setSqrValues(nextValues);
+        setXTurn(!xTurn);
+    }
+
+    let winner = theWinnerIs(sqrValues);
+    let messenge;
+    
+    if (winner) {
+        messenge = "The winner is: " + winner;
+    } else {
+        messenge = (xTurn ? "X" : "O") + " Turn";
+    }
+    console.log(messenge);
+
+    return (
+        <div>
+            <div className="messenge">{messenge}</div>
+            <div className="board-row">
+                <Square value={sqrValues[0]} onSquareClick={() => handleClick(0)}/>
+                <Square value={sqrValues[1]} onSquareClick={() => handleClick(1)}/>
+                <Square value={sqrValues[2]} onSquareClick={() => handleClick(2)}/>
+            </div>
+            <div className="board-row">
+                <Square value={sqrValues[3]} onSquareClick={() => handleClick(3)}/>
+                <Square value={sqrValues[4]} onSquareClick={() => handleClick(4)}/>
+                <Square value={sqrValues[5]} onSquareClick={() => handleClick(5)}/>
+            </div>
+            <div className="board-row">
+                <Square value={sqrValues[6]} onSquareClick={() => handleClick(6)}/>
+                <Square value={sqrValues[7]} onSquareClick={() => handleClick(7)}/>
+                <Square value={sqrValues[8]} onSquareClick={() => handleClick(8)}/>
+            </div>
+        </div>
+    );
+}
+
+function theWinnerIs(board) {
+    let lines = [
+        [0,1,2],
+        [3,4,5],
+        [6,7,8],
+        [0,3,6],
+        [1,4,7],
+        [2,5,8],
+        [0,4,8],
+        [2,4,6]
+    ];
+    for (let i =0; i< lines.length; i++) {
+        let [a,b,c] = lines[i];
+        if (board[a] && board[a] === board[b] && board[a] === board[c]) {
+            return board[a];
+        }
+    }
+    return null;
```

Comment:
A more functional approach prevents you from making a `off by one index` error
Your example is simple and it's fine to do it that way

But just so you know what I'm talking about, here's how to do it very functional and full of semantic variable names:

``` tsx
let winner = null;
lines.forEach((line)  => {
    const boardLineValues = line.map((index) => board[index]);
    const first = boardLineValues[0];
    if (first === null) {
        return;
    }

    const areAllEqual = boardLineValues.every(value => value === first)
    if (areAllEqual) {
        winner = first;
    }
})
return winner;
```

Reply:
done